File: matlab/calcstainresult.m
function [imageDecon] = calcstainresult(imageDecon, Imax, funContrast, varargin)
%UNTITLED Summary of this function goes here
%   Detailed explanation goes here

%% input layer
if ~exist('funContrast', 'var'), funContrast = []; end

%% reverse deconvolution
imageDecon = Imax* exp(-imageDecon);
imageDecon = Imax - uint8(imageDecon);

%% apply constrast stretching
if ~isempty(funContrast)

end

end

File: python/calcstainresult.py

import numpy as np

def calcstainresult(imageDecon, Imax, funContrast=[]):
        
    # reverse deconvolution
    imageDecon = Imax * np.exp(-imageDecon)
    imageDecon = np.clip(imageDecon, 0, 255)
    imageDecon = Imax - np.uint8(imageDecon)
    
    # apply constrast stretching
#    if funContrast == []:
        
    return imageDecon